provider/azure/internal/azureauth/discovery.go

```diff
@@ -29,7 +29,9 @@ type recordAuthHeaderPolicy struct {
 
 func (p *recordAuthHeaderPolicy) Do(req *policy.Request) (*http.Response, error) {
 	resp, err := req.Next()
-	p.authHeader = resp.Header.Get(authenticateHeaderKey)
+	if resp.Header != nil {
+		p.authHeader = resp.Header.Get(authenticateHeaderKey)
+	}
```

Comment:
@jack-w-shaw can you do a follow-up on just the resp being nil as well.

`if resp != nil && resp.Header != nil {`

Reply:
Yep

Merging this so we have a common base, to help avoid merge troubles

Soz should have mentioned in a comment here